### PR TITLE
[INLONG-10864][SDK] Transform SQL support md5 function

### LINKED ISSUE #10864: [Feature][SDK] Transform SQL support md5 function
### Description

- parent issue #10796 

MD5(string): Return the MD5 hash value of a string in the form of a 32-bit hexadecimal digit string; If the string is NULL, return NULL.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zkplo**
Fixes #10864 

### Motivation

Add one arithmetic function classe: Md5Function. Also, add the corresponding unit test codes.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Md5Function.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    public Md5Function(Function expr) {
+        msgParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object msgObj = msgParser.parse(sourceData, rowIndex, context);
+        if (msgObj == null)
+            return null;
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Md5Function.java:46]: add braces.

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — DISMISSED** [review]

**Review by vernedeng — DISMISSED** [review]

**Comment by aloyszhang** [comment]
plz rebase the master to resolve the conflict

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Md5Function.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object msgObj = msgParser.parse(sourceData, rowIndex, context);
+        if (msgObj == null) {
+            return null;
+        }
+        String msg = msgObj.toString();
+        if (msg.isEmpty()) {
+            return null;
+        }
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Md5Function.java:51]: It returns null when the input string is empty. However, the MD5 value for an empty string should be "d41d8cd98f00b204e9800998ecf8427e", it is not `NULL`.

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 3]
```diff
@@ -55,6 +55,39 @@ public class TestTransformArithmeticFunctionsProcessor {
...
     }
 
+    @Test
+    public void testMd5Function() throws Exception {
+        String transformSql = "select md5(numeric1) from source";
+        TransformConfig config = new TransformConfig(transformSql);
+        TransformProcessor<String, String> processor = TransformProcessor
+                .create(config, SourceDecoderFactory.createCsvDecoder(csvSource),
+                        SinkEncoderFactory.createKvEncoder(kvSink));
+        // case1: "1|4|6|8"
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:65]:
```suggestion
        // case1: md5(1)
```

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 4]
```diff
@@ -55,6 +55,39 @@ public class TestTransformArithmeticFunctionsProcessor {
...
+        List<String> output1 = processor.transform("1|4|6|8");
+        Assert.assertEquals(1, output1.size());
+        Assert.assertEquals("result=c4ca4238a0b923820dcc509a6f75849b", output1.get(0));
+
+        // case2: "-1|4|6|8"
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:70]: ditto.

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 5]
```diff
@@ -55,6 +55,39 @@ public class TestTransformArithmeticFunctionsProcessor {
...
+
+        // case4: "1|4|6|8"
+        transformSql = "select md5(numericxx) from source";
+        config = new TransformConfig(transformSql);
+        processor = TransformProcessor
+                .create(config, SourceDecoderFactory.createCsvDecoder(csvSource),
+                        SinkEncoderFactory.createKvEncoder(kvSink));
+        List<String> output4 = processor.transform("1|4|6|8");
+        Assert.assertEquals(1, output4.size());
+        Assert.assertEquals("result=null", output4.get(0));
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:88]: It is better to add a test case: md5("").

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java (hunk 3 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java]: done

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java (hunk 5 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java]: done

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java (hunk 4 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java]: done

**Review by aloyszhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by yfsn666 — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**